### PR TITLE
Add support for knocks

### PR DESCRIPTION
**Pull request by maltee1**
Please review

### CODE REVIEW

File: mautrix/bridge/matrix.py [hunk 1]
```diff
@@ -817,7 +835,12 @@ async def int_handle_event(self, evt: Event, send_bridge_checkpoint: bool = True
             prev_content = unsigned.prev_content or MemberStateEventContent()
             prev_membership = prev_content.membership if prev_content else Membership.JOIN
             if evt.content.membership == Membership.INVITE:
-                await self.int_handle_invite(evt)
+                if prev_membership == Membership.KNOCK:
+                    await self.handle_accept_knock(
+                        evt.room_id, UserID(evt.state_key), evt.event_id
+                    )
```
Comment (tulir) [inline mautrix/bridge/matrix.py:841]: This seems to be missing the sender and reason

File: mautrix/bridge/matrix.py [hunk 2]
```diff
@@ -865,6 +902,14 @@ async def int_handle_event(self, evt: Event, send_bridge_checkpoint: bool = True
...
                         evt.room_id, UserID(evt.state_key), evt.content, prev_content, evt.event_id
                     )
+            elif evt.content.membership == Membership.KNOCK:
+                await self.handle_knock(
+                    evt.room_id,
+                    UserID(evt.state_key),
+                    evt.sender,
+                    evt.content.reason,
+                    evt.event_id,
+                )
```
Comment (tulir) [inline mautrix/bridge/matrix.py:912]: This one doesn't need the sender (reason might still make sense)

File: mautrix/bridge/matrix.py [hunk 3]
```diff
@@ -243,6 +243,24 @@ async def handle_unban(
...
+
+    async def handle_reject_knock(
+        self, room_id: RoomID, user_id: UserID, event_id: EventID
+    ) -> None:
+        pass
+
+    async def handle_accept_knock(
+        self, room_id: RoomID, user_id: UserID, event_id: EventID
+    ) -> None:
+        pass
```
Comment (tulir) [inline mautrix/bridge/matrix.py:262]: The parameters here don't match what they're called with

**Review by tulir — CHANGES_REQUESTED** [review]

**Comment by maltee1** [comment]
merge?

**Pull request merged**